packages/itwin/measure-tools/src/ui-2.0/MeasureToolsUiProvider.tsx

```diff
@@ -32,11 +33,15 @@ export interface MeasureToolsUiProviderOptions {
   // If we check for sheet to 3d transformation when measuring in sheets
   enableSheetMeasurement?: boolean;
   stageUsageList?: string[];
+  // Callback that is invoked when a tracked feature is used.
+  onFeatureUsed?: (feature: Feature) => void;
 }
 
 export class MeasureToolsUiItemsProvider implements UiItemsProvider {
   public readonly id = "MeasureToolsUiItemsProvider";
-  private _props: RecursiveRequired<MeasureToolsUiProviderOptions>;
+  private _props: Omit<RecursiveRequired<MeasureToolsUiProviderOptions>, 'onFeatureUsed'> & {
+    onFeatureUsed?: (feature: Feature) => void;
+  };
```

Comment:
This is getting a bit confusing, especially since it doesn't seem like `this._props.onFeatureUsed` is used outside of the constructor: why should it be added to `this._props` at all? One idea is to each of the `_props` members into its own private member. Then we wouldn't have this unnecessary coupling between internal state and arguments.

Reply:
Thank you, onFeatureUsed is indeed used in constructor only. I will remove `& {
    onFeatureUsed?: (feature: Feature) => void;
  };` part and directly use `props?.onFeatureUsed` instead. This is also reasonable as it does not make a lot of sense to initialize onFeatureUsed with any default value if it is a member of `this._props`

```
this._props = {
      itemPriority: props?.itemPriority ?? 20,
      groupPriority: props?.groupPriority ?? 10,
      widgetPlacement: {
        location: props?.widgetPlacement?.location ?? StagePanelLocation.Right,
        section: props?.widgetPlacement?.section ?? StagePanelSection.Start,
      },
      enableSheetMeasurement: props?.enableSheetMeasurement ?? false,
      stageUsageList: props?.stageUsageList ?? [StageUsage.General],
      onFeatureUsed: props?.onFeatureUsed, // will also remove this line
    };
```
About the idea to each of the `_props` members into its own private member, I think we can do that in another PR where we can replace `_props` usage with those private properties. But does that mean we do not need `RecursiveRequired` introduced in this [commit](https://github.com/iTwin/viewer-components-react/commit/5dde769047e695c11eb1e91e930de28a267ea7ab) (to make sure all `_props` members initialized with at least some value) anymore?

Reply:
Sure, I was just thinking that perhaps someone would come along and do something similar for another arg that wasn't used outside of the constructor, but whichever you feel more comfortable doing. Just an observation. This repo/file isn't my code so don't want to impose.